src/editors/containers/ProblemEditor/components/SelectTypeModal/hooks.js

```diff
@@ -1,14 +1,50 @@
-import {
-  useState,
-} from 'react';
+import { useEffect, useState } from 'react';
 import { StrictDict } from '../../../../utils';
+import { ProblemTypeKeys, ProblemTypeOrder } from '../../../../data/constants/problem';
+import * as module from './hooks';
 
 export const state = StrictDict({
   selected: (val) => useState(val),
 });
 
-export const onSelect = ({ selected }) => {
-  return;
+export const selectHooks = () => {
+  const [ selected, setSelected ] = module.state.selected(ProblemTypeKeys.SINGLESELECT);
+  return {
+    selected,
+    setSelected,
+  };
 };
 
-export default { state, onSelect };
+export const onSelect = ( setProblemType, selected ) => () =>
+  setProblemType({ selected });
+
+export const useArrowNav = (selected, setSelected) => {
```

Comment:
I messed around with tests for this but have no idea how to do it.

Reply:
you'd have to add a property to document, which I think you can do like this:

```
it('should test something', () => {
  const spyFunc = jest.fn();
  Object.defineProperty(global.document, 'querySelector', { value: spyFunc });
  <run some test>
  expect(spyFunc).toHaveBeenCalled()
});
```